### DISCUSSION TITLE
Indexation issue

### DISCUSSION THREAD

**Discussion by jojobuilds** [vercel/community#6986] [Help]
### Summary

Hello everyone,

I'm working with a NextJS project on Vercel and facing a specific challenge with search engine indexing. I understand that Vercel automatically applies noindex to preview URLs, but it seems this does not extend to development URLs or custom production domains. My goal is to ensure that my production site (site.com) is indexed by Google, while my development environment (site.vercel.app) remains unindexed to prevent duplicate content issues.

Has anyone successfully navigated this situation? How can I configure my NextJS app and Vercel settings to ensure proper indexing for my production domain and exclude the development environment from search engines? Any guidance or tips would be greatly appreciated!

Thanks!

### Steps to Reproduce

As described in OP.

**Comment by versecafe** [comment]
@jojogotscammed You can check your Vercel environment with VERCEL_ENV and in a `robots.ts` set the indexing to allowed only for production, on top of this make sure your `sitemap.xml` or `sitemap.ts` only have your production domain in it. The last step is to go into the google search console, and verify your site with a meta tag or DNS TXT record so google can give you feedback about any issues with indexing and you can manually force it to fetch sitemaps or index certain pages. You can also go and make a bing webmasters account and import your data from google search to make sure everything is being indexed for Bing & DuckDuckGo

**Reply by jojobuilds** [reply]
Thank you! 

So, I added this into next.config.js:

`module.exports = {
env: {
PRODUCTION_URL: 'xyz.com',
DEVELOPMENT_URL: 'xyz.vercel.app'
},
};`

Then, I created:

robots.txt.js

```
export default function handler(req, res) {
const robotsTxt = process.env.VERCEL_ENV === 'production'
? `User-agent: *\nAllow: /`
: `User-agent: *\nDisallow: /`;
res.setHeader('Content-Type', 'text/plain');
res.write(robotsTxt);
res.end();
}
```

And next-sitemap.config.js:

```
module.exports = {
siteUrl: process.env.VERCEL_URL || 'https://xyz.com//',
generateRobotsTxt: true, // this will enable the generation of robots.txt
exclude: ['/api/*'], // exclude all API paths
robotsTxtOptions: {
additionalSitemaps: [
'https://xyz.com/sitemap.xml', // specify the sitemap path
],
},
};
```

But, to no avail. 

The robots.txt file looks like this: 

```
# *
User-agent: *
Allow: /

# Host
Host: xyz.vercel.app

# Sitemaps
Sitemap: xyz.vercel.app/sitemap.xml
Sitemap: https://xyz.com/sitemap.xml
```

I'm at a loss, because if I understood correctly this should be al dealth with automatically by Vercel, which makes total sense because this would be a incredibly common issue and equally 'hard' to fix like you see in my case...

Any clues how to reach Vercel so they can just block my preview Vercel domain? I don't have a paid plan and can't afford it.

**Reply by versecafe** [reply]
You can go declare canonical URLs inside the metadata too, and if you make and verify your google search console account you can go control what google indexes and delists
